client/src/App.tsx

```diff
@@ -1,12 +1,14 @@
-import { useState } from "react";
+import { FC, useState } from "react";
 import { Navigate, Outlet, Route, Routes } from "react-router-dom";
 import LoginForm from "./components/LoginForm";
+import Home from "./components/Home";
+import MapComponent from "./components/Map";
 import NavHeader from "./components/NavHeader";
 import NotFound from "./components/NotFound";
 import API from "./API/API";
-import "./styles/global.scss"
+import "./styles/global.scss";
 
-function App() {
+const App: FC = () => {
   const [loggedIn, setLoggedIn] = useState(false);
   const [user, setUser] = useState("");
```

Comment:
I strongly advise for defining a global user context to manage session data. 
You can check an [example](https://github.com/polito-WA1-2024-exam/exam-2-umberto-fontanazza/blob/main/client/src/App.jsx) in this project of mine but this is totally up to you.